### ISSUE TITLE
Navigating back to a collection view scrolls back to top

### ISSUE THREAD

**Issue by danielsaidi** [defagos/SwiftUICollection#5]
I noticed that if I navigate to a sheet (for now, since navigation links don't work) from a collection view and return back to the collection view, the collection view will scroll to the top.

I wonder if this has anything to do with what you wrote in your post, about disabling focus for performance reasons. Could it be that the button loses focus when you leave the screen, then don't automatically receives focus when navigating back, and that this cause the collection view to scroll to the top?

I will try and see if I can get navigation links to work by triggering a navigation link that is placed outside of the collection from cells within the collection view. Perhaps this focus problem doesn't happen then.

**Comment by defagos** [comment]
Yes, quite remind me of such issues, but focus is a tricky beast on tvOS and the issue could be elsewhere, I cannot really tell.

In our apps we present modal view controllers on top of our collections, and scroll position is correctly preserved when returning.

**Comment by danielsaidi** [comment]
With modal view controllers, do you mean SwiftUI `sheets` or `full screen covers`?

**Comment by danielsaidi** [comment]
I found that this strange behavior consisted of two different problems:

- Movies are not being focused when returning
- The collection scrolls to top when returning

My screens trigger a data fetch in `onAppear` and I found that I had commented out a line that checks if the screen already has data, which should abort the data fetch operation. However, with this line commented out, the screen refetched its data every time it appeared, which caused the collection to reload.

I restored the commented out line and that solved the scroll issue...plus the cell also retains its focus! Closing this 👍
